pkg/controller/authentication/pgsql.go

```diff
@@ -0,0 +1,221 @@
+      memory: {{ .MemLim }}
+    requests:
+      cpu: {{ .CpuReq }}
+      ephemeral-storage: {{ .ESReq }}
+      memory: {{ .MemReq }}
+`
+
+// Create Postgresql Cluster CR
+func (r *ReconcileAuthentication) createUpdateFromYaml(instance *operatorv1alpha1.Authentication, scheme *runtime.Scheme, yamlContent []byte) error {
+	obj := &unstructured.Unstructured{}
+	jsonSpec, err := yaml.YAMLToJSON(yamlContent)
+	if err != nil {
+		return fmt.Errorf("could not convert yaml to json: %v", err)
+	}
+
+	if err := obj.UnmarshalJSON(jsonSpec); err != nil {
+		return fmt.Errorf("could not unmarshal resource: %v", err)
+	}
+
+	obj.SetNamespace(instance.Namespace)
+
+	// Set CommonServiceConfig instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, obj, scheme); err != nil {
+		return err
+	}
+	err = r.client.Create(context.TODO(), obj)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			if err := r.client.Update(context.TODO(), obj); err != nil {
+				return fmt.Errorf("could not Update resource: %v", err)
+			}
+			return nil
+		}
+		return fmt.Errorf("could not Create resource: %v", err)
+	}
+
+	return nil
+}
+
+func (r *ReconcileAuthentication) newPgsqlServiceSpec(instance *operatorv1alpha1.Authentication) *operatorv1alpha1.PgsqlServiceSpec {
+	storageClass := instance.Spec.PgsqlService.StorageClass
+
+	if storageClass == "" {
+		storageclass, err := r.getstorageclass()
+		if err != nil {
+			return nil
+		} else {
+			instance.Spec.PgsqlService.StorageClass = storageclass
+
+		}
+	} else {
+		scExist := r.storageclassAvailabe(storageClass)
+		if !scExist {
+			storageclass, err := r.getstorageclass()
+			if err != nil {
+				return nil
+			} else {
+				instance.Spec.PgsqlService.StorageClass = storageclass
+
+			}
+		} else {
+			// provided storageclass is present in the cluster.
+		}
+	}
+	if instance.Spec.PgsqlService.LogLevel == "" {
+		instance.Spec.PgsqlService.LogLevel = "info"
+	}
+	cpuReq := instance.Spec.PgsqlService.Resources.Requests.Cpu().String()
+	if cpuReq == "" {
+		cpuReq = "75m"
+	}
+	memReq := instance.Spec.PgsqlService.Resources.Requests.Memory().String()
+	if memReq == "" {
+		memReq = "256Mi"
+	}
+	esReq := instance.Spec.PgsqlService.Resources.Requests.StorageEphemeral().String()
+	if esReq == "" {
+		esReq = "128Mi"
+	}
+	cpuLim := instance.Spec.PgsqlService.Resources.Limits.Cpu().String()
+	if cpuLim == "" {
+		cpuLim = "200m"
+	}
+	memLim := instance.Spec.PgsqlService.Resources.Limits.Memory().String()
+	if memLim == "" {
+		memLim = "768Mi"
+	}
+	esLim := instance.Spec.PgsqlService.Resources.Limits.StorageEphemeral().String()
+	if esLim == "" {
+		esLim = "512Mi"
+
+	}
+	return &operatorv1alpha1.PgsqlServiceSpec{LogLevel: instance.Spec.PgsqlService.LogLevel, StorageClass: instance.Spec.PgsqlService.StorageClass, ImageName: shatag.GetImageRef("POSTGRESQL_IMAGE"), CpuReq: cpuReq, MemReq: memReq, ESReq: esReq, CpuLim: cpuLim, MemLim: memLim, ESLim: esLim}
```

Comment:
Please format this so that everything is visible without having line wrap enabled.

Reply:
Will address in part 2 after we get it merged with operator-sdk v1.